### DISCUSSION TITLE
What's the point of USB grabber?

### DISCUSSION THREAD

**Discussion by paunovic** [awawa-dev/HyperHDR#713] [Q&A] [answered]
Hey everyone, first of all, I've discovered this project yesterday and it looks amazing! Great work.

I've been looking to install it on LG G2 TV, and everything makes sense except I don't understand what is the point of USB grabber.

From what I've deduced, it limits 4k throughput to 60Hz and I'd like to avoid this, i.e. retain 4k/120hz signal. Was thinking if just having 4x2 HDMI splitter would work, where I would feed one HDMI output to the TV, and other one directly to rpi/HyperHDR, this would avoid having USB grabber at all?

Alternative solution I was thinking of is to send image from TV somehow differently to HyperHDR, i.e. with external camera or try rooting my TV and installing an app that sends the signal.

But pretty sure I'm missing something obvious here, so any advice is welcome. Basically I'd like to retain all TV functionality while still having awesome ambiance experience.

Thanks!

**Comment by awawa-dev** [comment, marked as answer]
Hi
Well.. in past nobody expected the HDCP inquisition ;) 3 cents were added in Linux, partly by the producers of closed drivers, and partly by changes that were chosen in the architecture, where software capturing was the last worry, you get the picture of the situation. And since users have come to terms with it (apart from occasional revolutionary calls, e.g. to return to x11), there is no sign that anything will change soon. If you don't want to use hardware grabbers, you can try WIndows and the DirectX software grabber (the best capabilities and performance), it will work with 4k120 source but you won't win against HDCP. Moreover, if you are aiming for 4k120, the HDMI2.1 equipment is expensive and it's not only about grabbers, there are such solutions, but the price is simply very high.

**Reply by paunovic** [reply]
Thanks, seems like we're out of luck then. Are you saying that splitting HDMI output via 4x2 splitter to TV and HyperHDR won't work because of HDCP/DRM?

I'll ponder more on camera setup. Aside from it being inconvenient, i.e. having external camera, I think this might yield most results, as I altogether avoid having to deal with signal and HDCP.

**Reply by awawa-dev** [reply]
Some splitters can remove HDCP, the main problem is HDMI2.1(4k120): you need a splitter that can also down scale video to 1080p120 (not for the TV, but for a grabber, some are 1080p120 capable) or grabber that is 4k120 capable (if it does not have pass through you need a splitter, but all pass-through grabbers that I know wont allow capture if HDCP is detected...so splitter is needed)
